source/common/filesystem/file_shared_impl.cc

```diff
@@ -2,12 +2,14 @@
 
 #include <cstring>
 
+#include "common/common/utility.h"
+
 namespace Envoy {
 namespace Filesystem {
 
 Api::IoError::IoErrorCode IoFileError::getErrorCode() const { return IoErrorCode::UnknownError; }
 
-std::string IoFileError::getErrorDetails() const { return ::strerror(errno_); }
+std::string IoFileError::getErrorDetails() const { return errorDetails(errno_); }
```

Comment:
One issue we do have here is using FormatMessage on Windows produces an unexpected error string when a POSIX error code is returned. We could potentially change `errorDetails()` to detect error codes in the POSIX range and return the `strerror` value for those

Reply:
`//test/common/filesystem:filesystem_impl_test` expects the error string from EBADF (error code 9) but FormatMessage sees error code 9 (ERROR_INVALID_BLOCK) and returns "The storage control block address is invalid."

Reply:
We are going to resolve this in a subsequent PR that refactors filesystem impl to not use POSIX APIs on Windows to ensure a consistent experience